data/merge.go

```diff
@@ -8,17 +8,43 @@ import (
 )
 
 func setVal(p Point, v reflect.Value) {
-	switch v.Type().Kind() {
-	case reflect.String:
-		v.SetString(p.Text)
-	case reflect.Int:
-		v.SetInt(int64(p.Value))
-	case reflect.Float64, reflect.Float32:
-		v.SetFloat(p.Value)
+	t := v.Type()
+	k := t.Kind()
+	if !v.CanSet() {
+		log.Println("setVal failed, cannot set value")
```

Comment:
Should these return an error instead of printing to console?

Reply:
generally yes. The only reason I can think of to log an error and not return is there is some value in a partial merge.

Reply:
So going forward, `setVal` returns an error. The system will continue to merge points even if an error occurs, and the Decode function will return the first error that occurred.